docs/modules/ROOT/pages/how-to/monitor-http-endpoints.adoc

```diff
@@ -0,0 +1,262 @@
+    rules:
+    - alert: EndpointDown
+      expr: probe_success == 0 <1>
+      for: 5m <2>
+      labels:
+        severity: critical
+      annotations:
+        summary: "Endpoint {{ \$labels.instance }} down"
+YAML
+----
+<1> The `probe_success` metric is created by the Blackbox Exporter as an overall status of the probe.
+You can also alert on the individual metrics like `probe_http_status_code` or `probe_duration_seconds`.
+Query all available metrics with `{job=~"probe/.*"}`.
+<2> The `for` clause specifies how long the probe has to fail before the alert is triggered.
+Shorter times might help find problems earlier but might also trigger false positives or flukes.
+
+. Verify the alert was picked up by Prometheus
++
+Navigate to the OpenShift Web Console https://console.<CLUSTER>.cloud/dev-monitoring/ns/<APP_NAMESPACE>/alerts.
++
+image::monitoring/blackbox-alert-list.png[List of Blackbox alerts https://console.<CLUSTER>.cloud/dev-monitoring/ns/<APP_NAMESPACE>/alerts]
++
+The alert should show up in the list.
+If it doesn't show up see the <<troubleshooting>> section.
+
+== Troubleshooting [[troubleshooting]]
+
+=== No Entry is Shown in the List if Querying the `up` Metric `up{job=~"probe/.*"}`
+
+[INFO]
+The probe CRD isn't well-formed or Prometheus hasn't yet scraped the probe.
+
+Prometheus hasn't yet scraped the probe.
+Wait for three more minutes and check again.
+
+If no entry is shown in the list, the `Probe` CRD is most likely is misconfigured.
+Especially check the `spec.prober.url`.
+The format is `HOST:PORT`.
+`http://HOST` isn't valid.
+
+=== The Value of the `up` Metric is `0`
+
+[INFO]
+The probe CRD is well-formed but the blackbox exporter isn't reachable.
+
+If the value is `0`, the `Probe` CRD is valid, but the blackbox exporter can't be reached.
+Double check the `.spec.prober.url` of the `Probe` CRD.
+
+Check if the blackbox exporters deployment is running and shown as ready.
+
+=== `probe_success` shows `0`
+
+[INFO]
+The probe CRD is well-formed and the blackbox exporter is reachable.
+The target isn't reachable, the certificate is bad, or the target returns an unwanted status code.
+
+Check the `probe_http_status_code` for the job.
+Only `2XX` is considered a success after the final redirect.
+If the status code is `0`, the problem is earlier in the chain.
+
+=== No Alerts Shows Up in the Alert List
+
+// The promtheus rule is validated by the prometheus operator. I did not manage to apply a rule that was not valid.
+
+[INFO]
+The `PrometheusRule` CRD might not be well-formed and might have failed to apply.
+
+. Check if the `PrometheusRule` CRD was created successfully.
++
+[source,shell]
+----
+kubectl -n "${APP_NAMESPACE}" get prometheusrules
+----
+
+=== Debug Non-Successful Probes
+
+. Connect to the blackbox exporter
++
+[source,bash]
+----
+kubectl --namespace "${EXPORTER_NAMESPACE}" port-forward svc/blackbox-exporter-prometheus-blackbox-exporter 9115
+----
+
+. Open the blackbox exporter in a browser
```

Comment:
Genuine question: Can the blackbox exporter probe logs be checked with `kubectl logs` at all?

Reply:
Not in the default configuration at least. Should i invest some more time here to check if it can be enabled?

Reply:
I wouldn't spend too much time, but it might be worth it to check if there's an obvious option in blackbox exporter options.